R/utils_groupcomparison.R

```diff
@@ -439,3 +439,67 @@ getSamplesInfo = function(summarization_output) {
     result
 }
 
+#' Perform group comparison per protein in parallel
+#' @param summarized_list output of MSstatsPrepareForGroupComparison
+#' @param contrast_matrix contrast matrix
+#' @param save_fitted_models if TRUE, fitted models will be included in the output
+#' @param repeated logical, output of checkRepeatedDesign function
+#' @param samples_info data.table, output of getSamplesInfo function
+#' @param numberOfCores Number of cores for parallel processing. 
+#' A logfile named `MSstats_groupComparison_log_progress.log` is created to 
+#' track progress.
+#' @keywords internal
+.groupComparisonWithMultipleCores = function(summarized_list, contrast_matrix,
+                                             save_fitted_models, repeated, samples_info, 
+                                             numberOfCores) {
+    groups = colnames(contrast_matrix)
+    has_imputed = attr(summarized_list, "has_imputed")
+    all_proteins_id = seq_along(summarized_list)
+    cl = parallel::makeCluster(numberOfCores)
+    parallel::clusterExport(cl, c("MSstatsGroupComparisonSingleProtein", 
+                                  "contrast_matrix", "repeated", "groups", 
+                                  "samples_info", "save_fitted_models", "has_imputed"), 
+                            envir = environment())
```

Comment:
what's the meaning of envir = environment()?

Reply:
`environment()` refers to the R environment of the function that is executing these lines of code, i.e. `.groupComparisonWithMultipleCores`.  By default, `clusterExport` sets the environment to the global environment, but the global environment does not contain the function's variables.

I can save `environment()` as a variable prior to make the code more readable.